src/main.cpp

```diff
@@ -528,42 +527,46 @@ int main(int argc, char** argv) {
             !Global::config().has("generate")) {
         // ------- interpreter -------------
 
-        // configure interpreter
-        std::unique_ptr<Interpreter> interpreter;
-        if (!Global::config().has("interpreter")) {
-            interpreter = std::make_unique<LVM>(*ramTranslationUnit);
-        } else {
-            if (Global::config().get("interpreter") == "RAMI") {
-                interpreter = std::make_unique<RAMI>(*ramTranslationUnit);
-            } else {
-                interpreter = std::make_unique<LVM>(*ramTranslationUnit);
-            }
-        }
-
         std::thread profiler;
         // Start up profiler if needed
         if (Global::config().has("live-profile") && !Global::config().has("compile")) {
             profiler = std::thread([]() { profile::Tui().runProf(); });
         }
-        // execute translation unit
-        interpreter->executeMain();
-
-        // If the profiler was started, join back here once it exits.
-        if (profiler.joinable()) {
-            profiler.join();
-        }
 
-        // only run explain interface if interpreted
-        if (Global::config().has("provenance")) {
-            // construct SouffleProgram from env
-            InterpreterProgInterface interface(*interpreter);
-            if (Global::config().get("provenance") == "explain") {
-                explain(interface, false);
-            } else if (Global::config().get("provenance") == "explore") {
-                explain(interface, true);
+        // configure and execute interpreter
+        if (!Global::config().has("interpreter") || Global::config().get("Interpreter") == "LVM") {
```

Comment:
If you specify a default in the list of config options then you can just test for what the value is, rather than existence too.